### ISSUE TITLE
we lost the "early build" feature of the legacy app in the main branch

### ISSUE THREAD

**Issue by davehorton** [jambonz/jambonz-webapp#171]
Till now, the build process (by which I mean both the packer build of an AMI as well as the docker build) has done the "npm run build" step, which means that time to deploy is faster.  However, at run time we then have had to "fix up" the build by providing the jambonz base url, which is only known at deploy time.  

You can see how that is done 
- for kubernetes in the entrypoint.sh [here](https://github.com/jambonz/jambonz-webapp/blob/legacy/entrypoint.sh)
- for EC2 in the cloudformation script [here](https://github.com/jambonz/jambonz-infrastructure/blob/fd887d9ae4aab2f4a08bc7d4f45bb025062fbdcf/cloudformation/jambonz-mini.yaml#L207)

The main branch does not include this facility.  As a result, when deploying a system via cloudformation the webapp is non-functional until you log into the server and edit the jambonz-webapp/.env to put in the base url and then run `npm run build`.

@kitajchuk @EgleHelms We need the main branch to have this feature similar to how it used to work.  Because the build process is now different (using vite) I am not sure what changes are required but hope you can work them out.

**Comment by kitajchuk** [comment]
> Till now, the build process (by which I mean both the packer build of an AMI as well as the docker build) has done the "npm run build" step, which means that time to deploy is faster. However, at run time we then have had to "fix up" the build by providing the jambonz base url, which is only known at deploy time.
> 
> You can see how that is done
> 
> * for kubernetes in the entrypoint.sh [here](https://github.com/jambonz/jambonz-webapp/blob/legacy/entrypoint.sh)
> * for EC2 in the cloudformation script [here](https://github.com/jambonz/jambonz-infrastructure/blob/fd887d9ae4aab2f4a08bc7d4f45bb025062fbdcf/cloudformation/jambonz-mini.yaml#L207)
> 
> The main branch does not include this facility. As a result, when deploying a system via cloudformation the webapp is non-functional until you log into the server and edit the jambonz-webapp/.env to put in the base url and then run `npm run build` and then restart the newly-rebuilt webapp.
> 
> @kitajchuk @EgleHelms We need the main branch to have this feature similar to how it used to work. Because the build process is now different (using vite) I am not sure what changes are required but hope you can work them out.

There is nothing really different about the build process. I remember the old app had a dot env and the docker stuff. This has that too. How was the url that is only known at runtime handled before?

**Comment by davehorton** [comment]
well, in the old world we updated build/index.html at runtime and added a script tag
```
echo "REACT_APP_API_BASE_URL=http://$PUBLIC_IPV4/api/v1" > /home/admin/apps/jambonz-webapp/.env 
REACT_APP_API_BASE_URL=http://$PUBLIC_IPV4/api/v1 TAG="<script>window.JAMBONZ = { APP_API_BASE_URL: '$REACT_APP_API_BASE_URL'};</script>"
sed -i -e "\@</head>@i\ $TAG" /home/admin/apps/jambonz-webapp/build/index.html
```
In the new version, we now build in /dist rather than /build and in the .env we set `VITE_API_BASE_URL`

I guess we just need to update entrypoint.sh and the CF script to do the same thing -- insert a script section in the head of the index.html just changing the file to edit. Though, [based on this code](https://github.com/jambonz/jambonz-webapp/blob/93d313b21c9f488ce7f3b2074111504b398a0dfb/src/api/constants.ts#L25) I guess we need to change the name of the variable assignment in the script code as well?

**Comment by kitajchuk** [comment]
> well, in the old world we updated build/index.html at runtime and added a script tag
> 
> ```
> echo "REACT_APP_API_BASE_URL=http://$PUBLIC_IPV4/api/v1" > /home/admin/apps/jambonz-webapp/.env 
> REACT_APP_API_BASE_URL=http://$PUBLIC_IPV4/api/v1 TAG="<script>window.JAMBONZ = { APP_API_BASE_URL: '$REACT_APP_API_BASE_URL'};</script>"
> sed -i -e "\@</head>@i\ $TAG" /home/admin/apps/jambonz-webapp/build/index.html
> ```
> 
> In the new version, we now build in /dist rather than /build and in the .env we set `VITE_API_BASE_URL`
> 
> I guess we just need to update entrypoint.sh and the CF script to do the same thing -- insert a script section in the head of the index.html just changing the file to edit. Though, [based on this code](https://github.com/jambonz/jambonz-webapp/blob/93d313b21c9f488ce7f3b2074111504b398a0dfb/src/api/constants.ts#L25) I guess we need to change the name of the variable assignment in the script code as well?

The new web app does the same thing. It shouldn’t matter what the build directory is unless thats hardcoded somewhere else? The entrypoint:

https://github.com/jambonz/jambonz-webapp/blob/93d313b21c9f488ce7f3b2074111504b398a0dfb/entrypoint.sh#L14

**Comment by davehorton** [comment]
ah ok, it _is_ hardcoded in the cloudformation script, which I need to update in the same way.  Let me try that -- this is what I have been testing, so it makes sense that it is failing

**Comment by kitajchuk** [comment]
> ah ok, it _is_ hardcoded in the cloudformation script, which I need to update in the same way. Let me try that -- this is what I have been testing, so it makes sense that it is failing

Ok. If it’s easier I think we can specify the output directory for vite and set it back to build.

**Comment by davehorton** [comment]
No that's ok, lets leave as is.  I've updated the cloudformation yaml and am in the process of testing it now

**Comment by kitajchuk** [comment]
> No that's ok, lets leave as is. I've updated the cloudformation yaml and am in the process of testing it now

Coolio

**Comment by EgleHelms** [comment]
Hi @kitajchuk @davehorton seems like this does not require any action. Maybe this issue can be closed then?

**Comment by davehorton** [comment]
Yes, I have fixed and tested my CF as well as the packer build so I think we are good here.
